pkg/mimir/grpc_push_check_test.go

```diff
@@ -50,21 +66,133 @@ func TestGrpcInflightMethodLimiter(t *testing.T) {
 		})
 		require.Equal(t, 1, m.startCalls)
 		require.Equal(t, 1, m.finishCalls)
+
+		m.returnError = errors.New("hello there")
+		ctx, err = l.RPCCallStarting(context.Background(), ingesterPushMethod, nil)
+		require.Error(t, err)
+		_, ok := status.FromError(err)
+		require.True(t, ok)
+		require.Nil(t, ctx.Value(pushTypeCtxKey)) // Original context expected in case of errors.
+	})
+
+	t.Run("distributor push via httpgrpc", func(t *testing.T) {
+		m := &mockDistributorReceiver{}
+
+		l := newGrpcInflightMethodLimiter(nil, func() distributorPushReceiver { return m })
+
+		ctx, err := l.RPCCallStarting(context.Background(), "test", nil)
+		require.NoError(t, err)
+		l.RPCCallFinished(ctx)
+		require.Equal(t, 0, m.startCalls)
+		require.Equal(t, 0, m.finishCalls)
+
+		ctx, err = l.RPCCallStarting(context.Background(), httpgrpcHandleMethod, metadata.New(map[string]string{
+			httpgrpc.MetadataMethod:      "POST",
+			httpgrpc.MetadataURL:         api.PrometheusPushEndpoint,
```

Comment:
[nit] Would be nice to add a prefix to ensure it matches anyway.

Reply:
https://github.com/grafana/mimir/pull/6300/commits/e65144f2d4e0550167f294b66a61e740c261d230